pkg/storage/replica.go

```diff
@@ -2749,7 +2793,17 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 	r.store.raftEntryCache.addEntries(r.RangeID, rd.Entries)
 	r.mu.lastIndex = lastIndex
 	r.mu.raftLogSize = raftLogSize
-	r.mu.leaderID = leaderID
+	if r.mu.leaderID != leaderID {
+		r.mu.leaderID = leaderID
+		if r.mu.leaderID == r.mu.replicaID {
+			r.mu.proposalQuotaBase = r.mu.lastIndex
+			r.proposalQuota.reset(leaderProposalQuota)
+		} else {
+			r.proposalQuota.reset(followerProposalQuota)
```

Comment:
I'd just as soon eliminate the follower's use of a quota pool and the associated constant and just set the pool to nil on a leader -> follower transition, and allow followers to always propose and forward to the leader.